MainPage (인기 레시퍼, 추천 레시피, 추천 리뷰)기능구현

rkrzy opened the pull request.

rkrzy committed:
🚀 feat: MainPage 컨트롤러 추가

rkrzy committed:
🚀 feat: MainPageService 추가

rkrzy committed:
🔨 refactor: 함수추가

OnboardRepository

rkrzy committed:
🔨 refactor: Getter추가

Onboard

rkrzy committed:
🚀 feat: MainPageDTO 작성

rkrzy committed:
🚀 feat: 메인페이지 관련 리스트(인기레시퍼, 추천 레시피, 레시피 후기) 기능 추가

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageController.java:
```diff
@@ -0,0 +1,30 @@
+
+    MainPageService mainPageService;
+
+    @GetMapping("authentication")
```
경로에 "/"가 빠졌네용 아래도 마찬가지입니다.

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageController.java:
```diff
@@ -0,0 +1,30 @@
+@RequestMapping("/api/v1/main")
+public class MainPageController {
+
+    MainPageService mainPageService;
```
의존 주입이 안될 거 같은데 생성자 주입을 한번 알아보심이..

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,226 @@
+    static final int popularReciperCount = 3;
+    static final int popularReviewCount = 3;
+
+    RecipeRepository recipeRepository;
```
여기도 마찬가지로 의존 주입이 안될 거 같아요

rkrzy committed:
🔨 refactor: 코드양식 변경

로그인을 하지않고 메인을 접속했을때
- 추천 게시물을 찾는다면? -> 랜덤으로 6개 처리
- 리뷰를 받은게 하나도 없는데 리뷰평가를 받을려고 한다면?? -> 0반환

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+        private V value;
+    }
+
+    static final int recommendRecipeCount = 6;
```
이런 값들을 상수 선언하신 건 정말 탁원한 선택인 거 같아요.

보통 자바에서는 상수 선언을 대문자와 snake_case를 활용하긴 합니다!

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+
+    @Getter
+    @AllArgsConstructor
+    public class Pair<K, V> {
```
static inner 클래스에 대해서 한번 알아보시면 좋을 거 같아요

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+        return memberList.stream().map(Pair::getValue).collect(Collectors.toList());
+    }
+
+    private float averageRatingOfReview(List<Review> reviews) {
```
리뷰 값이 0.677777이라면 서버에서는 0.7로 응답해줘야 하지 않을까요?

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+        return classify;
+    }
+
+    private List<Recipe> bestRecipe(List<Recipe> recipes, List<Onboard> onboards, int recipeCount) {
```
recipeCount나 reciperCount 등 이미 위에서 static 상수 선언을 한 값들은 메서드 파라미터로 받기 보단 전역 상수로 메서드 내에서 직접 접근할 수도 있을 거 같아요!

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+        return findRandomValueList(reviews, reviewCount);
+    }
+
+    private <T> List<T> findRandomValueList(List<T> list, int Count) {
```
Count를 사용할 필요도 없지만(static 상수 떄문에) 네이밍은 소문자로 count로 하는 게 좋을 거 같아요!

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+        List<Onboard> memberOnboard = onboardRepository.findAllByMemberId(member.getId());
+        Map<Category, List<Onboard>> classify = new HashMap<>();
+        for (Onboard onboard : memberOnboard) {
+            if (classify.containsKey(onboard.getCategory().toString())) {
```
Map의 Key는 Category Enum인데 맵의 키를 검사할 때 toString() 을 하면 키를 못찾지 않을까요?

MinseokGo commented on src/main/java/grwm/develop/recipe/MainPageService.java:
```diff
@@ -0,0 +1,222 @@
+    private boolean findRecipeOfOnboard(Recipe recipe, List<Onboard> onboards) {
+        List<Hashtag> hashtags = hashtagRepository.findAllByRecipeId(recipe.getId());
+        for (Onboard onboard : onboards) {
+            if (recipe.getTitle().contains(onboard.getKeyword().toString())) {
```
이미 키워드는 String value 입니다!

MinseokGo commented on src/main/java/grwm/develop/recipe/dto/MainPageResponse.java:
```diff
@@ -0,0 +1,89 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
굿 너무 좋습니다

MinseokGo reviewed (commented):
너무 좋습니다 수정될 부분은 코멘트로 남겨놨으니 확인해보세용

rkrzy committed:
🔨 refactor: 코드수정

1. 의존성 주입 문제
- @Service가 없어서 controller에서 빈으로 인식을 못할수도 있을거 같음.
- private final키워드를 사용하지 않아서 문제가 발생할거 같음.

2. 경로에 "/"가 빠져있는 문제 해결

3. 매개변수를 없어지게 하고 상수를 대입해서 함수구조 변경

4, inner class를 사용할때 static키워드를 사용해야 하는 이유를 알았음.

5. toString으로 인식이 안될거 같은 문제 부분 변경

6. 소수점 두번째 자리에서 반올림해서 첫번째 자리까지만 출력
